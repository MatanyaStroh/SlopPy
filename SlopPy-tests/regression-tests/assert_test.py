# should ONLY taint 'ratio' as NA but leave other local vars alone
def foo(line):
  tokens = line.split()
  ratio = float(tokens[0]) / float(tokens[1])
  assert(ratio < 1.0)
  print tokens, ratio

foo('1 2')
foo('4 2')
foo('1 3')
foo('8 2')
foo('1 4')
