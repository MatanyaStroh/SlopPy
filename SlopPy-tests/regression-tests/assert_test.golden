{NA <type 'exceptions.AssertionError'> AssertionError()
 Traceback (most recent call last):
  File "regression-tests/assert_test.py", line 5, in foo
    assert(ratio < 1.0)

Locals: {'tokens': ['4', '2'], 'line': '4 2', 'ratio': 2.0}
}
{AssertionError: Set local var 'ratio' to NA | regression-tests/assert_test.py foo:5}
{NA <type 'exceptions.AssertionError'> AssertionError()
 Traceback (most recent call last):
  File "regression-tests/assert_test.py", line 5, in foo
    assert(ratio < 1.0)

Locals: {'tokens': ['8', '2'], 'line': '8 2', 'ratio': 4.0}
}
{AssertionError: Set local var 'ratio' to NA | regression-tests/assert_test.py foo:5}
